pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -78,6 +79,8 @@ protected void sampleAndCheckInterruption() {
     Tracing.ThreadAccountantOps.sampleMSE();
     if (Tracing.ThreadAccountantOps.isInterrupted()) {
       earlyTerminate();
+      throw QueryErrorCode.SERVER_RESOURCE_LIMIT_EXCEEDED.asException("Resource limit exceeded for operator: "
```

Comment:
Confirming that clients will get this error code if OOM protection kills a MSE query ? Does this then also fix #14999 ?